pkg/resources/server.go

```diff
@@ -275,6 +301,12 @@ func (rs *resourceServer) ListAndWatch(e *pluginapi.Empty, s pluginapi.DevicePlu
 		return err
 	}
 
+	err := rs.updateCDISpec()
```

Comment:
why do you prefer to create the spec here an not on allocate when required ?

Reply:
thinking on it, i think its fine to keep it here. that is, CDI spec generation is performed in listAndWatch